src/client.ts

```diff
@@ -62,19 +63,21 @@ export type RequestConfig = CustomClientOptions & {
  * The Skynet Client which can be used to access Skynet.
  */
 export class SkynetClient {
-  // TODO: This is currently the url of the skapp and not the portal. It should be the value of 'skynet-portal-api' header. This will be a promise, which will be a breaking change.
-  portalUrl: string;
+  // The value of the 'skynet-portal-api' header.
+  portalUrl: Promise<string>;
```

Comment:
So what about this one, this is really awkward.. I can get over it but can we then at least document explicitly why it has to be like this. It's very very unexpected for the `constructor` to be initiating anything asynchronous. That's just not done.

Reply:
This is from the other PR.